### PR TITLE
MarginPool farming

### PR DESCRIPTION
**Pull request by haythem96**
# Task: Feature Name:

MarginPool farming

## High Level Description

This PR implemented the internal balances accounting and add the excess balance farming functionality.

### Code

- [x] Unit test 100% coverage
- [x] Does your code follow the naming and code documentation guidelines?

### Documentation

- [x] Is your code up to date with the spec? 
- [x] Have you added your tests to the testing doc?

### CODE REVIEW

File: contracts/MarginPool.sol [hunk 1]
```diff
@@ -130,4 +159,39 @@ contract MarginPool {
...
+
+    /**
+     * @notice function to set farmer address
+     * @dev can only be called by MarginPool owner
+     * @param _farmer farmer address
+     */
+    function setFarmer(address _farmer) external onlyOwner {
+        emit FarmerUpdated(farmer, _farmer);
+
+        farmer = _farmer;
```
Comment (antoncoding) [inline contracts/MarginPool.sol:195]:
```suggestion
        farmer = _farmer;

        emit FarmerUpdated(farmer, _farmer);

```

File: test/marginPool.test.ts [hunk 2]
```diff
@@ -373,6 +373,84 @@ contract('MarginPool', ([controllerAddress, user1, random]) => {
...
+
+    it('should set farmer address when called from owner', async () => {
+      await marginPool.setFarmer(farmer, {from: owner})
+
+      assert.equal(await marginPool.farmer(), farmer, 'farmer address mismatch')
+    })
+
+    it('should revert farming when receiver address is equal to zero', async () => {
+      // send more usdc to pool
+      await usdc.mint(marginPool.address, new BigNumber('100'))
```
Comment (antoncoding) [inline test/marginPool.test.ts:404]: I think it's better to move this line into `before()` block

**Review by antoncoding — APPROVED** [review]
LGTM
Can be merged after adding test for farming amount > external balance - storedBalcne 
https://coveralls.io/jobs/67334430/source_files/2186019993#L180

File: contracts/MarginPool.sol [hunk 3]
```diff
@@ -130,4 +159,40 @@ contract MarginPool {
...
+     * @notice function to collect excess balance
+     * @dev can only be called by farmer address
+     * @param _asset asset address
+     * @param _receiver receiver address
+     * @param _amount amount to harvest
+     */
+    function farm(
+        address _asset,
+        address _receiver,
+        uint256 _amount
```
Comment (aparnakr) [inline contracts/MarginPool.sol:173]: why even take an amount? Just pay out the difference between stored and external balance?

File: contracts/MarginPool.sol (hunk 1 above)
Comment (antoncoding) [reply contracts/MarginPool.sol]: can we just simplified it to `FarmerUpdated(farmer)` ?

Comment (aparnakr) [reply contracts/MarginPool.sol]: its nice to know the old and new tbh

**Review by aparnakr — APPROVED** [review]
LGTM

**Comment by haythem96** [comment]
> LGTM
> Can be merged after adding test for farming amount > external balance - storedBalcne
> https://coveralls.io/jobs/67334430/source_files/2186019993#L180

Added.

**Pull request merged**